sky/core.py

```diff
@@ -953,6 +956,99 @@ def storage_ls() -> List[Dict[str, Any]]:
     return storages
 
 
+def storage_refresh() -> None:
+    """Syncs the internal status and the external status of storages
+
+    Removes storages from the table in ~/.sky/state.db that are managed
+    by Sky but no longer exist in the cloud console. It also adds storages
+    to the same table if they are managed by Sky, exist in the cloud console,
+    but are not currently listed in ~/.sky/state.db.
+    """
+    green = colorama.Fore.GREEN
+    red = colorama.Fore.RED
+    bold = colorama.Style.BRIGHT
+    reset = colorama.Style.RESET_ALL
+
+    internal_storage_list = storage_ls()
+    internal_buckets_status = collections.defaultdict(set)
+    for storage in internal_storage_list:
+        for store in storage['store']:
+            internal_buckets_status[store].add(storage['name'])
+
+    storage_enabled_clouds = global_user_state.get_storage_enabled_clouds()
+    external_buckets_status = collections.defaultdict(set)
+
+    # sync state.db and external status of storages
+    for storage_enabled_cloud in storage_enabled_clouds:
+        storetype: storage_lib.StoreType
+        store_class: Type[storage_lib.AbstractStore]
+        # TODO: use get_storetype_from_cloud when
+        # storage_enabled_clouds are officially maintained
```

Comment:
Can we use `get_storetype_from_cloud` now?

Reply:
Not yet. I meant to say when this [issue](https://github.com/skypilot-org/skypilot/issues/1943) is resolved.